src/compiler/watch.ts

```diff
@@ -302,6 +302,10 @@ namespace ts {
         // There is no extra check needed since we can just rely on the program to decide emit
         const builder = createBuilder({ getCanonicalFileName, computeHash });
 
+        if (watchingHost.system.clearScreen) {

```

Comment:
Can you create function expression clearScreen when entering this function and use that instead of if checks here and in update program?
```typescript
const clearScreen = watchingHost.system.clearScreen ? watchingHost.system.clearScreen.bind(watchingHost.system) : noop;
```